src/graphnet/models/data_representation/graphs/nodes/nodes.py

```diff
@@ -469,3 +469,184 @@ def _construct_nodes(self, x: torch.Tensor) -> torch.Tensor:
             graph[:event_length, idx] = x[ids, self.feature_indexes[feature]]
 
         return graph
+
+
+class DOMSummaryFeatures(NodeDefinition):
```

Comment:
As you point out in the docstring, the method may aggregate information across any defined cluster. To avoid confusion, I think the name of the class should reflect this. It could for example be `ClusterSummaryTheo` :-)